src/diffusers/models/unet_2d_blocks.py

```diff
@@ -558,7 +660,7 @@ def __init__(
 
         # there is always at least one resnet
         resnets = [
-            ResnetBlock2D(
+            get_resnet_block(
```

Comment:
I think it might be better to just append the `ResnetBlock2D` here than introduce the block fetching function.

Reply:
I'm aware that some of the blocks use a mix, i.e. the ones Kandinsky used for MOVQ: `DownEncoderBlock2D`, `AttnDownEncoderBlock2D`, `UpDecoderBlock2D`, `AttnUpDecoderBlock2D`. 

We could maybe just use `get_resnet_block` for these blocks instead - I would prefer that way because it is more explicit and readable. I used the function everywhere for now because:
1. Technically all these blocks can be configured to use both types, I don't have the complete knowledge to know if it would break existing models.
2. Just use this as an example to ensure backward compatibility. I only split `ResnetBlock2D` into 2 blocks here in the draft but maybe we want to have more blocks with more specific configurations. It can become unmanageable without a block-fetching function 

Overall I agree with you it is better to just use the appropriate Resnet blocks, but I will wait to hear more about:
1.  feedbacks on how many blocks we want to split  into
2. what's the best strategy to ensure backward compatibility, given that we might not have complete data on how these blocks are used

Reply:
> feedbacks on how many blocks we want to split into

I think we'd want to split into the most common block types. I think this number is still quite unclear. 

> We could use get_resnet_block for these blocks instead - I would prefer that way too, because it is more explicit and readable. 

How about we use the fetcher function but also add a comment above to denote which type of ResNet block is being used?